### PR TITLE
Gitter backend auto configuration

### LINKED ISSUE #369: Build BOT_IDENTITY using multiple env vars
This is a good template for our `config.py` : https://github.com/rroemhild/docker-errbot/blob/master/config.py#L103

```
BOT_IDENTITY = {}

# username
if 'BOT_USERNAME' in os.environ:
    BOT_IDENTITY['username'] = os.environ['BOT_USERNAME']

# password
if 'BOT_PASSWORD' in os.environ:
    BOT_IDENTITY['password'] = os.environ['BOT_PASSWORD']

# server
if 'BOT_SERVER' in os.environ:
    if ':' in os.environ['BOT_SERVER']:
        server, port = os.environ['BOT_SERVER'].split(':')
        BOT_IDENTITY['server'] = (server, int(port))
    else:
        BOT_IDENTITY['server'] = os.environ['BOT_SERVER']

# token
if 'BOT_TOKEN' in os.environ:
    BOT_IDENTITY['token'] = os.environ['BOT_TOKEN']

# endpoint
if 'BOT_ENDPOINT' in os.environ:
    BOT_IDENTITY['endpoint'] = os.environ['BOT_ENDPOINT']

# nickname
if 'BOT_NICKNAME' in os.environ:
    BOT_IDENTITY['nickname'] = os.environ['BOT_NICKNAME']

# port
if 'BOT_PORT' in os.environ:
    BOT_IDENTITY['port'] = os.environ['BOT_PORT']

# ssl
if 'BOT_SSL' in os.environ:
    BOT_IDENTITY['ssl'] = os.environ['BOT_SSL']
```

Needs to also keep backwards compatibility for `COBOT_TOKEN`, and update the readme.

### PR DESCRIPTION
**Pull request by jayvdb**
Closes #369

Related to #354

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -3,9 +3,46 @@
...
+    BOT_IDENTITY['token'] = os.environ.get('COBOT_TOKEN')
 
-BOT_EXTRA_BACKEND_DIR = os.path.join(COBOT_ROOT, 'err-backend-gitter')
+if BOT_IDENTITY['server'] and ':' in BOT_IDENTITY['server']:
+    server, port = os.environ['BOT_SERVER'].split(':')
+    BOT_IDENTITY['server'] = (server, int(port))
+
+BACKEND = os.environ.get('BACKEND')
+if not BACKEND:
+    if BOT_IDENTITY['token']:
```
Comment (Vamshi99) [inline config.py]: According to line 22, this statement will be always true

**Comment by meetmangukiya** [comment]
ack fe5bb1e0d010ea2c1fa40d25909ebbe0779275e9 25526dd1cf358080ef7d45cb84caca34982f1595

**Comment by meetmangukiya** [comment]
@jayvdb looks good, rebase and merge 👍

**Comment by meetmangukiya** [comment]
@gitmate-bot rebase

**Comment by meetmangukiya** [comment]
reack https://github.com/coala/corobo/pull/484/commits/af65ccf02a8c45d53352eb546aa990507328757c https://github.com/coala/corobo/pull/484/commits/15e54943bfd9f42b22ef6aa9e2d6fa0c17005830

**Pull request merged**